Snippets update

gpokat opened the pull request:
~~Added explanatory comment about math package usage~~
Changed Main() entry point to Run() as per design and toolchain

This small update of front page code snippets will add
explanatory comment to highlight that provided Carbon code
is  hypothetical and meant to show the look and feel of the language.

Also it delivers change of Main() to Run() to
highlight correct entry point for Carbon lang.

gpokat committed:
Snippets update

Added explanatory comment about math package usage
Changed Main() entry point to Run() as per design and toolchain

danakj commented on docs/design/README.md:
````diff
@@ -151,7 +151,7 @@ Here is a simple function showing some Carbon code:
 ```carbon
 import Core library "io";
-import Math;
+import Math; //hypothetical package. Do not exist yet.
````
```suggestion
// This is a hypothetical package for demonstrating imports.
import Math;
```

danakj replied:
(carbon does not have trailing comments)

gpokat replied:
addressed.
use leading, thanks

danakj replied:
There's no plan to make it exist, so I don't think we should say "yet". I provided what I think is a more accurate comment.

gpokat replied:
@danakj sry that I do not fully caught proposed intent of change.
Addressed now.
please re-review.
thanks.

danakj commented on docs/images/snippets.md:
```diff
@@ -84,7 +84,7 @@ auto main() -> int {
 // Carbon:
 package Geometry;
-import Math;
+import Math; //hypothetical package. Do not exist yet.
```
```suggestion
// This is a hypothetical package for demonstrating imports.
import Math;
```

gpokat replied:
addressed.
use leading, thanks

danakj commented on docs/images/snippets.md:
```diff
@@ -98,7 +98,7 @@ fn PrintTotalArea(circles: [Circle]) {
 }
 
-fn Main() -> i32 {
+fn Run() -> i32 { //The design and the toolchain now use Run()
```
```suggestion
fn Run() -> i32 {
```

gpokat replied:
addressed.
use leading, thanks

danakj replied:
I would prefer to not introduce this comment at all, as I suggested in my changes

danakj commented on docs/images/snippets.md:
```diff
@@ -120,7 +120,7 @@ struct Circle {
 package Geometry;
 import Cpp library "circle.h";
-import Math;
+import Math; //hypothetical package. Do not exist yet.
```
```suggestion
// This is a hypothetical package for demonstrating imports.
import Math;
```

gpokat replied:
addressed.
use leading, thanks

danakj reviewed (commented):
Thanks, a few suggestions

gpokat committed:
Snippets update

Switch trailing to leading comments

danakj reviewed (commented):
Further comments left inline

gpokat committed:
Snippets update

Use proper comments

danakj reviewed (approved):
LGTM

danakj commented:
Sorry, had a discussion offline where we noticed that there's a lot more about these examples that are not going to compile (e.g. the array syntax has changed, and has been renamed, Print doesn't exist). It might be better to say something before the code snippets instead of inside them. Under this heading: https://github.com/carbon-language/carbon-lang/blob/trunk/README.md#carbon-and-c Maybe "The Carbon code here is hypothetical and meant to show the look and feel of the language."?

The Main->Run code change remains worth doing still.

gpokat commented:
> Sorry, had a discussion offline where we noticed that there's a lot more about these examples that are not going to compile (e.g. the array syntax has changed, and has been renamed, Print doesn't exist). It might be better to say something before the code snippets instead of inside them. Under this heading: https://github.com/carbon-language/carbon-lang/blob/trunk/README.md#carbon-and-c Maybe "The Carbon code here is hypothetical and meant to show the look and feel of the language."?
> 
> The Main->Run code change remains worth doing still.

gotcha, will add clarification as per suggestion.
Another question, will it worth to add few more trivial snippets which fully compiled and runs for comparison c++ vs carbon ? To be able copy-paste and check right away ?
I have some at local

danakj commented:
> > Sorry, had a discussion offline where we noticed that there's a lot more about these examples that are not going to compile (e.g. the array syntax has changed, and has been renamed, Print doesn't exist). It might be better to say something before the code snippets instead of inside them. Under this heading: https://github.com/carbon-language/carbon-lang/blob/trunk/README.md#carbon-and-c Maybe "The Carbon code here is hypothetical and meant to show the look and feel of the language."?
> > The Main->Run code change remains worth doing still.
> 
> gotcha, will add clarification as per suggestion. Another question, will it worth to add few more trivial snippets which fully compiled and runs for comparison c++ vs carbon ? To be able copy-paste and check right away ? I have some at local

I think that everything is moving a little fast, it would mean we have to keep fixing those (as you've found here). I think it's probably best to leave things hypothetical for now on the front page. Our test suites demonstrate what is and isn't working so far.

gpokat committed:
Snippets update

Removed proposed comments from snippets
Added explanation to carbon-and-c heading

gpokat commented:
> > > Sorry, had a discussion offline where we noticed that there's a lot more about these examples that are not going to compile (e.g. the array syntax has changed, and has been renamed, Print doesn't exist). It might be better to say something before the code snippets instead of inside them. Under this heading: https://github.com/carbon-language/carbon-lang/blob/trunk/README.md#carbon-and-c Maybe "The Carbon code here is hypothetical and meant to show the look and feel of the language."?
> > > The Main->Run code change remains worth doing still.
> > 
> > 
> > gotcha, will add clarification as per suggestion. Another question, will it worth to add few more trivial snippets which fully compiled and runs for comparison c++ vs carbon ? To be able copy-paste and check right away ? I have some at local
> 
> I think that everything is moving a little fast, it would mean we have to keep fixing those (as you've found here). I think it's probably best to leave things hypothetical for now on the front page. Our test suites demonstrate what is and isn't working so far.

Please check if the additions are good now.
Thanks.

danakj reviewed (approved):
This LGTM, thanks

The change in docs/images/snippets.md won't actually regenerate the image files. There are instructions in the google docs folder that snippets.md links to at the top. Would you like to do that? If so, please go ahead and add the updated image to the PR. Otherwise, I can follow up with doing so.

gpokat commented:
> This LGTM, thanks
> 
> The change in docs/images/snippets.md won't actually regenerate the image files. There are instructions in the google docs folder that snippets.md links to at the top. Would you like to do that? If so, please go ahead and add the updated image to the PR. Otherwise, I can follow up with doing so.

yeah i will take a look and update this PR later after working day :)

~~Need an access to address a change on google drawing.~~
~~Requested.~~

Can access it via own copy.
PR updated with new snippet image.
Ready to merge.

gpokat committed:
Snippets update

CarbonInfraBot fix for text length

gpokat committed:
Snippets update

Updated carbon_snippet.svg with Run() entry point

gpokat committed:
Merge branch 'trunk' into udpateExamples

danakj commented:
I see the image is updated now, thanks!
